osquery/tables/system/linux/memory_info.cpp

```diff
@@ -0,0 +1,81 @@
+/*
+ *  Copyright (c) 2014-present, Facebook, Inc.
+ *  All rights reserved.
+ *
+ *  This source code is licensed under the BSD-style license found in the
+ *  LICENSE file in the root directory of this source tree. An additional grant
+ *  of patent rights can be found in the PATENTS file in the same directory.
+ *
+ */
+
+#include <string>
+
+#include <boost/algorithm/string.hpp>
+
+#include <osquery/core.h>
+#include <osquery/tables.h>
+#include <osquery/filesystem.h>
+
+#include "osquery/core/conversions.h"
+
+namespace osquery {
+namespace tables {
+
+const std::string memInfoPath = {"/proc/meminfo"};
+
+QueryData getMemoryInfo(QueryContext& context) {
+  QueryData results;
+  Row r;
+  r["memory_total"] = INTEGER(-1);
+  r["memory_free"] = INTEGER(-1);
+  r["buffers"] = INTEGER(-1);
+  r["cached"] = INTEGER(-1);
+  r["swap_cached"] = INTEGER(-1);
+  r["active"] = INTEGER(-1);
+  r["inactive"] = INTEGER(-1);
+  r["swap_total"] = INTEGER(-1);
+  r["swap_free"] = INTEGER(-1);
+
+  std::string meminfo_content;
+  if (forensicReadFile(memInfoPath, meminfo_content).ok()) {
+    // Able to read meminfo file, now grab info we want
+    for (const auto& line : split(meminfo_content, "\n")) {
+      std::vector<std::string> tokens;
+      boost::split(
+          tokens, line, boost::is_any_of("\t "), boost::token_compress_on);
+      if (line.find("MemTotal:") == 0) {
```

Comment:
Instead of writing a large set of conditionals here, you can:
```cpp
std::map<std::string, std::string> indexes = {
  {"memory_total", "MemTotal"},
  {"memory_free", "MemFree"},
};
```

Then:
```cpp
for (const auto& p : indexes) {
  if (line.find(p.second) == 0) {
    r[p.first] = ...;
    break;
  }
}
```